lib/copy_peste/core/utils.rb

```diff
@@ -4,14 +4,30 @@ module Utils
 
       module_function
 
-      # Dynamically load a Ruby file which implements CopyPeste module standard
+      # Trigger the call of method defined by 'with' on defined object 'on'
+      # with parameter the child that inherited defined 'baseclass'.
       #
-      # @param dir [String] Absolute path of the module to be load.
-      # @param file [String] File name of the module to be load.
-      # @return [Object] Instance of the loaded module.
-      def load_module (dir, file)
-        loaded_mod = ModuleLoading::Loader.load File.join dir, file
-        loaded_mod.__cp_init__
+      # @param callback [Symbol] callback triggering
+      # @param on [Class] baseclass to watch callback on
+      # @param to [Object] target object
+      # @param with [Symbol] target object's method name
+      def hook_method(callback, on:, to:, with:)
+        hook = Module.new
+        hook.send(:define_method, callback) { |child|
+          super child
+          to.send with, child
+        }
+
+        on.extend hook
+        nil
```

Comment:
It's the best way to ensure your method is actually not returning a private value, nor a private function value, nor a value that make no sense to consider as a success flag.